mypy/nodes.py

```diff
@@ -443,11 +443,11 @@ class OverloadedFuncDef(FuncBase, SymbolNode, Statement):
 
     def __init__(self, items: List['OverloadPart']) -> None:
         super().__init__()
-        assert len(items) > 0
         self.items = items
         self.unanalyzed_items = items.copy()
         self.impl = None
-        self.set_line(items[0].line)
+        if len(items) > 0:
+            self.set_line(items[0].line)
```

Comment:
Maybe try using `self.impl` to set line, like in the `name()` method just below?

Reply:
That does not work, as `self.impl` is `None` at this point (see line above).

Reply:
OK, I see, maybe then set line in `deserialize()` method below? For example under the `if data.get('impl') is not None:` branch if `items` is empty (with a comment explaining that if overload items are empty then the line will not be set in `__init__`).